panel/src/panel/upload.js

```diff
@@ -234,6 +236,12 @@ export default (panel) => {
 				return this.done();
 			}
 
+			this.running = true;
+
+			const bulkFiles = [];
+			let page = 0;
+			let counter = 0;
```

Comment:
I think we could get this part a bit cleaner.

Chunking the files with something like (untested, but as general idea):

```js
const chunks = []:
while (chunks.length * 5 < this.files.length) {
  chunks.push(this.files.slice(this.chunks.length * 5, 5));
}
```

Also was wondering if https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of could help, but maybe not.

Reply:
I will look into this. But to be honest JS is not my strongest language :) So I'm thankful getting some feedback on this.